berdlhub/api_utils/spark_utils.py

```diff
@@ -212,15 +239,43 @@ async def start_spark_cluster(self, spawner) -> str:
         """
         username = spawner.user.name
         try:
-            self.logger.info(f"Creating Spark cluster for user {username}")
-            response = await self.create_cluster()
+            # Get the profile slug from the spawner
+            profile_slug = self._get_profile_slug_from_spawner(spawner)
+            self.logger.info(f"Creating Spark cluster for user {username} with profile '{profile_slug}'")
+
+            # Get cluster configuration from the profile
+            cluster_config = ClusterDefaults.from_profile(profile_slug)
+
+            # Create cluster with profile-specific configuration
+            response = await self.create_cluster(
+                worker_count=cluster_config.worker_count,
+                worker_cores=cluster_config.worker_cores,
+                worker_memory=cluster_config.worker_memory,
+                master_cores=cluster_config.master_cores,
+                master_memory=cluster_config.master_memory,
+            )
 
             master_url = getattr(response, "master_url", None)
             if not master_url:
                 raise SparkClusterError(f"Master URL not found in response: {response}")
 
             self.logger.info(f"Spark cluster created with master URL: {master_url}")
+
+            # Set cluster configuration as environment variables for the notebook
             spawner.environment["SPARK_MASTER_URL"] = master_url
+            spawner.environment["SPARK_WORKER_COUNT"] = str(cluster_config.worker_count)
```

Comment:
What happens when the user creates a spark cluster with the API, and these no longer apply? Shouldn't these be based on the running spark cluster?